src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -20,7 +20,17 @@ namespace Microsoft.NetCore.Analyzers.Runtime
     /// <summary>
     /// Detect the use of [RequiresPreviewFeatures] in assemblies that have not opted into preview features
     /// </summary>
-    public abstract class DetectPreviewFeatureAnalyzer : DiagnosticAnalyzer
+    public abstract class DetectPreviewFeatureAnalyzer<
+        TBaseTypeOrImplementsOrInherits,
+        TBaseTypeDeclarationSyntax,
+        TTypeConstraintSyntax,
+        TTypeArgumentList,
+        TParameterSyntax> : DiagnosticAnalyzer
+        where TBaseTypeOrImplementsOrInherits : SyntaxNode
+        where TBaseTypeDeclarationSyntax : SyntaxNode
+        where TTypeConstraintSyntax : SyntaxNode
+        where TTypeArgumentList : SyntaxNode
+        where TParameterSyntax : SyntaxNode
```

Comment:
I might miss something, seems this and `TBaseTypeDeclarationSyntax`  are not used